public/js/spa.js

```diff
@@ -463,3 +406,47 @@ $(document).on('loaded', function() {
   })(window.angular);
 
 });
+
+function MainController($scope, $route, $routeParams, $location, $http) {
+  $scope.$route = $route;
+  $scope.$location = $location;
+  $scope.$routeParams = $routeParams;
+  $scope.models = window.__models;
+  $scope.title = "Melody!!!1";
+  $scope.footer = "Very Info";
+  $scope.flashes = [];
+
+  $scope.$on('FlashMessage', function(event, args) {
+    //Flash.create(args.FlashType, args.FlashMsg, 'customAlert');
+    $scope.flashes.push({type:args.FlashType, text:args.FlashMsg});
+    console.log("Flash triggered");
+  });
+
+  $scope.action = MainControllerAction;
+}
+
+function MainControllerAction(targetProperty, modelName, action, id, data) {
+  if(!window.__models[modelName]) return;
+
+  var url = '/' + plural(modelName) + '/' + action + '/';
+
+  if (id) {
+    url += id;
+  }
+  console.log('action');
+  console.log(url);
+
+  $http({
+    method: 'get',
+    url: url,
+    params: data
+  }).then(
+    function(response) {
+      //console.log(response.data);
+      $scope[targetProperty] = response.data;
```

Comment:
should move it back into the controller for now, or convert $scope.action to call a shared service function